pylsp/workspace.py

```diff
@@ -152,6 +153,9 @@ def _progress_begin(
         percentage: Optional[int] = None,
     ) -> str:
         token = str(uuid.uuid4())
+
+        self._endpoint.request(self.M_INITIALIZE_PROGRESS, {'token': token}).result(_utils.CALL_TIMEOUT)
```

Comment:
I'm not sure why would you use a timeout for this specific request. Any compliant client should either respond with an OK or error response.

Though if you are not checking for client capability then all bets are off but that would be a server issue then.

Reply:
valid point, I removed the timeout.